src/main/java/com/marklogic/spark/writer/MarkLogicDataWriter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2023 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.marklogic.spark.writer;
+
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.DatabaseClientFactory;
+import com.marklogic.client.document.DocumentWriteSet;
+import com.marklogic.client.document.TextDocumentManager;
+import com.marklogic.client.io.Format;
+import com.marklogic.client.io.StringHandle;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.connector.write.DataWriter;
+import org.apache.spark.sql.connector.write.WriterCommitMessage;
+
+import java.io.IOException;
+import java.util.Map;
+
+public class MarkLogicDataWriter implements DataWriter<InternalRow> {
+    Map<String, String> map;
+    public MarkLogicDataWriter(Map<String, String> map) {
+        this.map = map;
+    }
+
+    @Override
+    public void write(InternalRow record) throws IOException {
+        System.out.println("************ Reached MarkLogicDataWriter.write ****************");
+      // TODO: Find a way to write based on the schema passed in by the user
+        DatabaseClient db = DatabaseClientFactory.newClient(map.get("spark.marklogic.client.host"), Integer.valueOf(map.get("spark.marklogic.client.port")),
```

Comment:
Use the line of code that's in `ReadContext` since we already have a good way of creating this:

    DatabaseClient client = DatabaseClientFactory.newClient(propertyName -> properties.get("spark." + propertyName));

I imagine we'll make a `WriteContext` to carry about state related to writing documents, but no need for that yet.